code/modules/mob/living/basic/slime/slime.dm

```diff
@@ -0,0 +1,385 @@
+
+	//AI related traits
+
+	/// Instructions you can give to slimes
+	var/static/list/pet_commands = list(
+		/datum/pet_command/idle,
+		/datum/pet_command/free,
+		/datum/pet_command/follow,
+		/datum/pet_command/point_targeting/attack,
+	)
+
+	/// Our evolve action
+	var/datum/action/innate/slime/evolve/evolve_action
+	/// Our reproduction action
+	var/datum/action/innate/slime/reproduce/reproduce_action
+
+/mob/living/basic/slime/Initialize(mapload, new_type=/datum/slime_type/grey, new_life_stage=SLIME_LIFE_STAGE_BABY)
+
+	. = ..()
+
+	evolve_action = new (src)
+	evolve_action.Grant(src)
+
+	reproduce_action = new (src)
+	reproduce_action.Grant(src)
+
+	set_slime_type(new_type)
+	set_life_stage(new_life_stage)
+	update_name()
+	regenerate_icons()
+
+	set_nutrition(SLIME_STARTING_NUTRITION)
+
+	AddComponent(/datum/component/buckle_mob_effect,  mob_effect_callback = CALLBACK(src, PROC_REF(feed_process)))
+	AddComponent(/datum/component/health_scaling_effects, min_health_slowdown = 2)
+	AddComponent(/datum/component/obeys_commands, pet_commands)
+
+	AddElement(/datum/element/ai_retaliate)
+	AddElement(/datum/element/basic_health_examine, light_damage_message = "It has some punctures in its flesh!", heavy_damage_message = span_bold("It has severe punctures and tears in its flesh!"), heavy_threshold = 0.4)
+	AddElement(/datum/element/footstep, footstep_type = FOOTSTEP_MOB_SLIME)
+	AddElement(/datum/element/soft_landing)
+	AddElement(/datum/element/swabable, CELL_LINE_TABLE_SLIME, CELL_VIRUS_TABLE_GENERIC_MOB, 1, 5)
+
+	ADD_TRAIT(src, TRAIT_CANT_RIDE, INNATE_TRAIT)
+	ADD_TRAIT(src, TRAIT_VENTCRAWLER_ALWAYS, INNATE_TRAIT)
+
+	RegisterSignal(src, COMSIG_HOSTILE_PRE_ATTACKINGTARGET, PROC_REF(on_slime_pre_attack))
+	RegisterSignal(src, COMSIG_ATOM_ATTACK_HAND, PROC_REF(on_attack_hand) )
+
+	ai_controller.set_blackboard_key(BB_SLIME_FACE, SLIME_MOOD_NONE)
+	ai_controller.set_blackboard_key(BB_SLIME_EVOLVE, evolve_action)
+	ai_controller.set_blackboard_key(BB_SLIME_REPRODUCE, reproduce_action)
+
+/mob/living/basic/slime/Destroy()
+
+	UnregisterSignal(src, list(COMSIG_HOSTILE_PRE_ATTACKINGTARGET, COMSIG_ATOM_ATTACK_HAND))
+
+	QDEL_NULL(evolve_action)
+	QDEL_NULL(reproduce_action)
+
+	return ..()
+
+///Random slime subtype
+/mob/living/basic/slime/random/Initialize(mapload, new_colour, new_life_stage)
+	. = ..(mapload, pick(subtypesof(/datum/slime_type)), prob(50) ? SLIME_LIFE_STAGE_ADULT : SLIME_LIFE_STAGE_BABY)
+
+///Friendly docile subtype
+/mob/living/basic/slime/pet
+	hunger_disabled = TRUE
+
+/mob/living/basic/slime/pet/Initialize(mapload, new_colour, new_life_stage)
+	. = ..()
+	set_pacified_behaviour()
+
+//Hilbert subtype
+/mob/living/basic/slime/hilbert/Initialize(mapload, new_colour, new_life_stage)
+	. = ..(mapload, /datum/slime_type/bluespace)
+	ai_controller?.set_blackboard_key(BB_SLIME_RABID, TRUE)
+
+//Overriden base procs
+
+/mob/living/basic/slime/adjust_nutrition(change)
+	..()
+	nutrition = min(nutrition, SLIME_MAX_NUTRITION)
```

Comment:
Also, you will need to add an override for `set_nutrition` to do a similar clamp.